### PR TITLE
Service for DQA secondary reports

### PR DESCRIPTION
**Pull request by l-k-**

### CODE REVIEW

File: services/etl/main.py [hunk 1]
```diff
@@ -123,9 +123,9 @@ def parse_commit(self):
 
 
 pedsnet_v2 = ETLConventionsResource(
-        document_url='https://api.github.com/repos/PEDSnet/Data_Models/contents/PEDSnet/V2/docs/Pedsnet_CDM_V2_OMOPV5_ETL_Conventions.md',  # noqa
+        document_url='https://api.github.com/repos/PEDSnet/Data_Models/contents/PEDSnet/docs/Pedsnet_CDM_ETL_Conventions.md',  # noqa
         commits_url='https://api.github.com/repos/PEDSnet/Data_Models/commits',  # noqa
-        file_path='PEDSnet/V2/docs/Pedsnet_CDM_V2_OMOPV5_ETL_Conventions.md')
+        file_path='PEDSnet/docs/Pedsnet_CDM_ETL_Conventions.md')
```
Comment (bruth) [inline services/etl/main.py:128]: Please remove changes to this file since it doesn't apply to this PR and PR #26 addresses this issue.

Comment (l-k-) [inline services/etl/main.py:128]: thanks, I didn't mean to include this, didn't notice that the file got pulled in.

**Comment by bruth** [comment]
A few observations:
- There are a lot of requests being made to the GitHub just to produce a single output on our end; up to 120 requests (sites \* tables)
- The code would need to be updated every time a new version of the analysis is performed, e.g. ETLv6
- The analyses are tied to particular versions of the data model 1.0.0 vs. 2.0.0, so that may be valuable information to extract
- Tables change across versions. For example from 1.0.0 to 2.0.0, [6 tables were added; 2 were removed](http://data-models.origins.link/compare/pedsnet/1.0.0/pedsnet/2.0.0)

To remove the need to do regular updates as the above things change, a more dynamic approach needs to be taken. My suggestion is to have the service clone the directory locally and scan the `SecondaryReports` directory and build up the set of sites, ETL versions, and tables per analysis. This is what the [data models service](https://github.com/chop-dbhi/data-models-service/blob/master/parse.go#L469-L533) does. Since the `SecondaryReports` directory has structure, you can build the tree based on that. The alternate is to ignore the structure and rely on the contents of the files themselves. The `DQA Version` value in every row uniquely identifies the analysis the result should be a apart of, e.g. `pedsnet-v1-Boston-ETLv1` &rarr; `<mode>-<version>-<site>-<analysis>`. So you could dynamically build a tree of analyses that could be aggregated (as you are doing now).

As far as the HTTP endpoints go, they can also be dynamic, e.g. `/pedsnet/:analysis` where `:analysis` would match on `ETLvN` if it exists in your indexed tree. As new analyses are performed, the service will just periodically pull down changes in the repo (or use a GitHub webhook to be notified of an update) so it will be up-to-date with the repo.

**Comment by l-k-** [comment]
That does make sense. Thank you!

**Pull request closed without merging**